VCAC-A/script/setup-docker.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+install_docker_engine()
+{
+    http_proxy="$1"
+    https_proxy="$2"
+    no_proxy="$3"
+    dnss="$4"
+
+    check_docker=`docker images`
+    if [ $? != 0 ];then
+        echo "Installing docker......" 
+        apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+	curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+	add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" 
+	export LC_ALL=C
+        apt-get update && apt-get install -y docker-ce docker-ce-cli containerd.io
+
+	proxy_config=/etc/systemd/system/docker.service.d/http-proxy.conf
+        if [ ! -f $proxy_config ]; then
+	    mkdir -p $(dirname $proxy_config)
+	    echo "[Service]" >> $proxy_config
+	    echo "Environment=\"http_proxy=$http_proxy\" \"https_proxy=$https_proxy\" \"no_proxy=$no_proxy\"" >> $proxy_config
+	    systemctl daemon-reload
+            systemctl restart docker
+        fi
+	config_file=~/.docker/config.json
+	if [ ! -f $config_file ]; then 
+	    mkdir -p $(dirname "$config_file")
+            cat > "$config_file" <<EOF
+{
+    "proxies": {
+        "default": {
+            "httpProxy": "$http_proxy", 
+            "httpsProxy": "$https_proxy",
+            "noProxy": "$no_proxy"
+        }
+    }
+}
+EOF
+        fi
+        
+        daemon_json=/etc/docker/daemon.json
+        if [ ! -f $daemon_json ]; then
+	        echo "{\"dns\":[$dnss]}" >> $daemon_json
+            mkdir -p $(dirname daemon_json)
+            systemctl restart docker
+        fi
+    fi
+}
+
+case "$0" in 
+    *install*)
+        install_docker_engine "$@"
+        ;;
+    *setup*)
+        sudo yum -y install NetworkManager
+        systemctl start NetworkManager
+        http_proxy=${http_proxy:-${http_proxy}}
+        https_proxy=${https_proxy:-${https_proxy}}
+        no_proxy=${no_proxy:-${no_proxy}}
```

Comment:
These do not make sense, set no_proxy to (if no_proxy is empty or null, return no_proxy, else no_proxy)

Reply:
CAPS missing. They should be:

        http_proxy=${http_proxy:-${HTTP_PROXY}}
        https_proxy=${https_proxy:-${HTTPS_PROXY}}
        no_proxy=${no_proxy:-${NO_PROXY}}